src/Chapter3.hs

```diff
@@ -971,6 +1129,30 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday deriving(Show, Enum, Eq)
+
+class Weekdays a where
+  isWeekend :: a -> Bool
+  nextDay :: a -> a
+  daysToParty :: a -> Int
+
+instance Weekdays Weekday where
+  isWeekend :: Weekday -> Bool
+  isWeekend Saturday = True
+  isWeekend Sunday = True
+  isWeekend _ = False
+
+  nextDay :: Weekday -> Weekday
+  nextDay Sunday = Monday
+  nextDay a = succ a
+
+  daysToParty :: Weekday -> Int
+  daysToParty day = go 0 day
+    where
+      go acc Friday = acc
+      go acc wday = go (acc + 1) (nextDay wday)
```

Comment:
That's correct implementation! Though, your probably can implement it efficiently using `fromEnum`